torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,435 @@
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class _SampleSequenceDict(TypedDict):
+    """Sample sequence dictionary."""
+
+    id: str
+    seq_id: list[int]
+
+
+class DigitalTyphoonAnalysis(NonGeoDataset):
+    """Digital Tyhphoon Dataset for Analysis Task.
+
+    This dataset contains typhoon-centered images, derived from hourly infrared channel
+    images captured by meteorological satellites. It incorporates data from multiple
+    generations of the Himawari weather satellite, dating back to 1978. These images
+    have been transformed into brightness temperatures and adjusted for varying
+    satellite sensor readings, yielding a consistent spatio-temporal dataset that
+    covers over four decades.
+
+    See `the Digital Typhoon website
+    <http://agora.ex.nii.ac.jp/digital-typhoon/dataset/>`_
+    for more information about the dataset.
+
+    Dataset features:
+
+    * infrared channel images from the Himawari weather satellite (512x512 px)
+      at 5km spatial resolution
+    * auxiliary features such as wind speed, pressure, and more that can be used
+      for regression or classification tasks
+    * 1,099 typhoons and 189,364 image
+
+    Dataset format:
+
+    * hdf5 files containing the infrared channel images
+    * .csv files containing the metadata for each image
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.20783/DIAS.664
+
+    .. versionadded:: 0.6
+    """
+
+    valid_tasks = ["classification", "regression"]
+    aux_file_name = "aux_data.csv"
+
+    valid_features = [
+        "year",
+        "month",
+        "day",
+        "hour",
+        "grade",
+        "lat",
+        "lng",
+        "pressure",
+        "wind",
+        "dir50",
+        "long50",
+        "short50",
+        "dir30",
+        "long30",
+        "short30",
+        "landfall",
+        "intp",
+    ]
+
+    url = "https://huggingface.co/datasets/torchgeo/digital_typhoon/resolve/main/WP.tar.gz{0}"  # noqa: E501
+
+    md5sums = {
+        "aa": "3af98052aed17e0ddb1e94caca2582e2",
+        "ab": "2c5d25455ac8aef1de33fe6456ab2c8d",
+    }
+
+    data_root = "WP"
+
+    def __init__(
+        self,
+        root: str = "data",
+        task: str = "regression",
+        features: Sequence[str] = ["wind"],
+        target: list[str] = ["wind"],
+        sequence_length: int = 3,
+        min_feature_value: Optional[dict[str, float]] = None,
+        max_feature_value: Optional[dict[str, float]] = None,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Digital Typhoon Analysis dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            task: whether to load "regression" or "classification" labels
```

Comment:
COWC has separate classes for classification and regression. But those datasets are marketed as two datasets. Not sure what is best here.

Reply:
The idea here is that either you can:
a) take the wind speeds for example and regard them as continuous regression targets
b) or split the wind speed range into bins with each bin being a different class